apis/src/components/atoms/schedule_controls.rs

```diff
@@ -0,0 +1,90 @@
+use crate::common::ScheduleAction;
+use crate::components::atoms::date_time_picker::DateTimePicker;
+use crate::providers::ApiRequests;
+use crate::responses::ScheduleResponse;
+use chrono::{DateTime, Duration, Local, Utc};
+use leptos::*;
+use uuid::Uuid;
+
+#[component]
+pub fn GameDateControls(player_id: Uuid, schedule: ScheduleResponse) -> impl IntoView {
+    let game_id = store_value(schedule.game_id);
+    let start_date = schedule.start_t;
+    let agreed = schedule.agreed;
+    let is_expired_or_none = start_date
+        .map(|t| t - Utc::now())
+        .map_or(true, |t| t.num_milliseconds() < 0);
+    let formated_game_date = move |time: DateTime<Utc>| {
+        let to_date = time - Utc::now();
+
+        let agreed_str = if agreed { "To play" } else { "Proposed" };
+        format!(
+            "{} in {} days, {} hours, {} minutes ({})",
+            agreed_str,
+            to_date.num_days(),
+            to_date.num_hours() % 24,
+            to_date.num_minutes() % 60,
+            time.with_timezone(&Local).format("%m-%d %H:%M")
+        )
+    };
+    let propose = Callback::from(move |date| {
+        let api = ApiRequests::new();
+        api.schedule_action(ScheduleAction::Propose(date, game_id.get_value()));
+    });
+    let accept = Callback::from(move |_| {
+        let api = ApiRequests::new();
+        api.schedule_action(ScheduleAction::Accept(game_id.get_value()));
+    });
+    let selected_time = RwSignal::new(Utc::now() + Duration::minutes(10));
+    view! {
+        <div class="flex justify-center px-2">
+            <Show when=move || is_expired_or_none>
+                <DateTimePicker
+                    text=""
+                    min=Local::now() + Duration::minutes(10)
+                    max=Local::now() + Duration::weeks(12)
+                    success_callback=Callback::from(move |utc| {
+                        selected_time.set(utc);
+                    })
+
+                    failure_callback=Callback::from(|_| { logging::log!("Failure to set date") })
```

Comment:
I would get rid of these logging callbacks, we can make the failure callback optional on <DateTimePicker/> and then it can be called with or without a failure one.